dfastmi/batch/AnalyserAndReporterWaqua.py

```diff
@@ -96,180 +86,282 @@ def analyse_and_report_waqua(
-        interest only.
+        Arguments
+        ---------
+        display : bool
+            Flag indicating text output to stdout.
+        report : TextIO
+            Text stream for log file.
+        reduced_output : bool
+            Flag to indicate whether WAQUA output should be reduced to the area of
+            interest only.
+        tstag : float
+            Number of days that the river is stagnant.
+        discharges : Vector
+            Array of discharges; one for each forcing condition.
+        apply_q : BoolVector
+            A tuple of 3 flags indicating whether each value should be used or not.
+        ucrit : float
+            Critical flow velocity [m/s].
+        old_zmin_zmax : bool
+            Specifies the minimum and maximum should follow old or new definition.
+        """
+        self.display = display
+        self.report = report
+        self.reduced_output = reduced_output
+        self.tstag = tstag
+        self.discharges = discharges
+        self.apply_q = apply_q
+        self.ucrit = ucrit
+        self.old_zmin_zmax = old_zmin_zmax
+        
+    def analyse(self, fraction_of_year : Vector, rsigma : Vector) -> tuple[bool, OutputDataWaqua]:
+        """
+        Read data from samples files exported from WAQUA simulations and performanalysis.
 
-    Returns
-    -------
-    dzq : numpy.ndarray
-        Array containing equilibrium bed level change.
-    firstm : int
-        Minimum M index read (0 if reduced_output is False).
-    firstn : int
-        Minimum N index read (0 if reduced_output is False).
-    """
-    cblok = str(stage)
-    if display:
-        ApplicationSettingsHelper.log_text("input_xyz", dict={"stage": stage, "q": q})
-        ApplicationSettingsHelper.log_text("---")
-        ApplicationSettingsHelper.log_text("")
-
-    discriptions = ApplicationSettingsHelper.get_text("file_descriptions")
-    quantities = ["velocity-zeta.001", "waterdepth-zeta.001", "velocity-zeta.002"]
-    files = []
-    for i in range(3):
-        if display:
-            ApplicationSettingsHelper.log_text("input_xyz_name", dict={"name": discriptions[i]})
-        cifil = "xyz_" + quantities[i] + ".Q" + cblok + ".xyz"
-        if display:
-            print(cifil)
-        if not os.path.isfile(cifil):
-            ApplicationSettingsHelper.log_text("file_not_found", dict={"name": cifil})
-            ApplicationSettingsHelper.log_text("file_not_found", dict={"name": cifil}, file=report)
-            ApplicationSettingsHelper.log_text("end_program", file=report)
+        Arguments
+        ---------
+        fraction_of_year : Vector
+            Fraction of year represented by each forcing condition.
+        rsigma : Vector
+            Array of relaxation factors; one per forcing condition.
+
+        Returns
+        -------
+        success : bool
+            Flag indicating whether analysis could be carried out.
```

Comment:
me too, but since this is used throughout the batch_mode_core like this and passed back into gui I didnt not want to take the risk and change more on this, since then the core and the gui would also be affected by this change.

Documentation regarding this has been reused from the analyse_and_report_waqua method.

Reply:
I will see If I can look into making this succes clearer in some way.

Reply:
removed it  up untill the analyse_and_report_waqua method